pallets/uniques/src/lib.rs

```diff
@@ -0,0 +1,754 @@
+		/// - `item`: The item of the item to be frozen.
+		///
+		/// Emits `Frozen`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(7)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::freeze())]
+		pub fn freeze(origin: OriginFor<T>, collection: T::CollectionId, item: T::ItemId) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::freeze(origin, collection, item)
+		}
+
+		/// Re-allow unprivileged transfer of an item.
+		///
+		/// Origin must be Signed and the sender should be the Freezer of the `collection`.
+		///
+		/// - `collection`: The collection of the item to be thawed.
+		/// - `item`: The item of the item to be thawed.
+		///
+		/// Emits `Thawed`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(8)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::thaw())]
+		pub fn thaw(origin: OriginFor<T>, collection: T::CollectionId, item: T::ItemId) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::thaw(origin, collection, item)
+		}
+
+		/// Disallow further unprivileged transfers for a whole collection.
+		///
+		/// Origin must be Signed and the sender should be the Freezer of the `collection`.
+		///
+		/// - `collection`: The collection to be frozen.
+		///
+		/// Emits `CollectionFrozen`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::freeze_collection())]
+		pub fn freeze_collection(origin: OriginFor<T>, collection: T::CollectionId) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::freeze_collection(origin, collection)
+		}
+
+		/// Re-allow unprivileged transfers for a whole collection.
+		///
+		/// Origin must be Signed and the sender should be the Admin of the `collection`.
+		///
+		/// - `collection`: The collection to be thawed.
+		///
+		/// Emits `CollectionThawed`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(10)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::thaw_collection())]
+		pub fn thaw_collection(origin: OriginFor<T>, collection: T::CollectionId) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::thaw_collection(origin, collection)
+		}
+
+		/// Change the Owner of a collection.
+		///
+		/// Origin must be Signed and the sender should be the Owner of the `collection`.
+		///
+		/// - `collection`: The collection whose owner should be changed.
+		/// - `owner`: The new Owner of this collection. They must have called
+		///   `set_accept_ownership` with `collection` in order for this operation to succeed.
+		///
+		/// Emits `OwnerChanged`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(11)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::transfer_ownership())]
+		pub fn transfer_ownership(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			owner: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::transfer_ownership(origin, collection, owner)
+		}
```

Comment:
@ETeissonniere since we use the ownership details only at the time an item is being burnt or a collection is being destroyed, it would work for us without any changes. In other words we rely on the ownership from the substrate uniques. When it's transferred, it's transferred for us too.